app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/MosaicImplicits.scala

```diff
@@ -113,127 +230,77 @@ class MosaicImplicits[HistStore: HistogramStore](histStore: HistStore)
               IO.raiseError(NoDataInRegionException)
             } else IO.unit
           }
-          filtered = BacksplashMosaic.filterRelevant(self)
-          extent = BacksplashImage.tmsLevels(z).mapTransform.keyToExtent(x, y)
+          overviewStream = self zip {
```

Comment:
I'm trying to follow the flow of the app here -- it looks like this is _always_ going to try and fetch an overview and then use an exception to control program flow?

I'm not sure of a better way to organize this for comprehension, but I'd be all for trying to break this specific set of "if there's an overview, try and get it, if not render how we normally do" into its own function or something to make it easier to follow what's happening.

Reply:
Does 15d4297 help?

Reply:
Yeah, that's great